lib/src/builder_implementation.dart

```diff
@@ -2254,17 +2322,22 @@ class _SuperclassFixedPoint extends FixedPoint<ClassElement> {
   /// class used as a mixin cannot have other superclasses than [Object]).
   /// TODO(eernst) implement: When mixins can have nontrivial superclasses
   /// we may or may not wish to enforce the bounds even for mixins.
-  Iterable<ClassElement> successors(ClassElement element) sync* {
+  Future<Iterable<ClassElement>> successors(ClassElement element) async {
```

Comment:
It seems to me we could benefit from this method returning a FutureOr.
Only a benchmark could tell how significant the issue is.

Reply:
Noted that in #182 as well.